pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -421,9 +421,18 @@ def evaluation_forward(self, model, batch, batch_idx, dataloader_idx, test_mode:
         # single GPU data transfer
         if self.single_gpu:
             # for single GPU put inputs on gpu manually
-            root_gpu = 0
+
             if isinstance(self.data_parallel_device_ids, list):
                 root_gpu = self.data_parallel_device_ids[0]
+
+                # set cuda device to root gpu
+                root_device = (torch.device("cuda", root_gpu)
+                               if root_gpu >= 0 else torch.device("cpu"))
+                torch.cuda.set_device(root_device)
+            else:
+                raise RuntimeError(
+                    'Expected `data_parallel_device_ids` as a list, cannot determine root gpu.'
```

Comment:
Above we have `if self.single_gpu`, so why should device ids be a list?

Reply:
There was already an if clause:`if isinstance(self.data_parallel_device_ids, list) `, I raised the runtime error as a safenet because `root_gpu = 0` was used earlier by default.